src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -853,86 +835,17 @@ instance Buildable AccountsSnapshot where
-            unzip <$> cloneAccounts accId Deleted addRemovedAccount
-        (oldExisting, newExisting) <-
-            unzip <$> cloneAccounts accId Existing addWAddress
-        let oldAddrMeta =
-                AccountsSnapshot
-                {asExisting = oldExisting, asDeleted = oldDeleted}
-            newAddrMeta =
-                AccountsSnapshot
-                {asExisting = newExisting, asDeleted = newDeleted}
-        logDebug $
-            sformat
-                ("Cloned wallet accounts: "%build%"\n\t-> "%build)
-                oldAddrMeta
-                newAddrMeta
-        return (oldAddrMeta, newAddrMeta)
-  where
-    noWMeta = InternalError "Can't get wallet meta (inconsistent db)"
-    cloneAccounts oldAccId lookupMode addToDB = do
-        accAddrs <- getAccountAddrsOrThrow lookupMode oldAccId
-        forM accAddrs $ \accAddr@CWAddressMeta {..} -> do
-            newAcc <- rederiveAccountAddress newSK newPass accAddr
-            _ <- addToDB newAcc
-            return (accAddr, newAcc)
-
-moveMoneyToClone
-    :: WalletWebMode m
-    => SendActions m
-    -> PassPhrase
-    -> CId Wal
-    -> [CWAddressMeta]
-    -> [CWAddressMeta]
-    -> m ()
-moveMoneyToClone sa oldPass wid oldAddrMeta newAddrMeta = do
-    let ms = WalletMoneySource wid
-    dist <-
-        forM (zip oldAddrMeta newAddrMeta) $ \(oldAcc, newAcc) ->
-            (cwamId newAcc, ) <$> getWAddressBalance oldAcc
-    whenNotNull dist $ \dist' ->
-        unless (all ((== mkCoin 0) . snd) dist) $
-        void $
-        sendMoney sa oldPass ms dist'
-
 changeWalletPassphrase
     :: WalletWebMode m
-    => SendActions m -> CId Wal -> PassPhrase -> PassPhrase -> m ()
-changeWalletPassphrase sa wid oldPass newPass = do
+    => CId Wal -> PassPhrase -> PassPhrase -> m ()
+changeWalletPassphrase wid oldPass newPass = do
     oldSK <- getSKByAddr wid
 
-    -- 'cloneWalletSetWithPass' will work badly if accounts / addresses ids
-    -- don't actually change
     unless (isJust $ checkPassMatches newPass oldSK) $ do
         newSK <- maybeThrow badPass $ changeEncPassphrase oldPass newPass oldSK
-
+        deleteSK oldPass
         addSecretKey newSK
-        oldAddrMeta <- (`E.onException` deleteSK newPass) $ do
-            (oldAddrMeta, newAddrMeta) <- cloneWalletSetWithPass newSK newPass wid
-                `E.onException` do
-                    logError "Failed to clone wallet"
-            moveMoneyToClone sa oldPass wid (asExisting oldAddrMeta) (asExisting newAddrMeta)
-                `E.onException` do
-                    logError "Money transmition to new wallet failed"
-                    mapM_ totallyRemoveWAddress $ asDeleted <> asExisting $ newAddrMeta
-            return oldAddrMeta
-        mapM_ removeWAddress $ asExisting oldAddrMeta
         setWalletPassLU wid =<< liftIO getPOSIXTime
-        deleteSK oldPass
   where
     badPass = RequestError "Invalid old passphrase given"
     deleteSK passphrase = do
```

Comment:
https://issues.serokell.io/issue/CSM-319
^ Created a separate issue to get rid of such comparisons